android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -109,18 +117,36 @@
             doneButton.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    addOnDoneAction(referenceEditText);
+                    addOnDoneAction(referenceEditText, doneButton, widgetArgs);
                 }
             });
         }
 
         ((JsonApi) context).addFormDataView(referenceEditText);
-        setViewTags(rootLayout);
-        prepareViewChecks(rootLayout, context);
+        setViewTags(rootLayout, widgetArgs);
+        prepareViewChecks(rootLayout, context, widgetArgs);
 
         return views;
     }
 
+    /**
+     * Returns the object that holds the repeating group count
+     *
+     * @return
+     * @throws JSONException
+     */
+    private JSONObject getRepeatingGroupCountObj(WidgetArgs widgetArgs) throws JSONException {
+        JSONObject repeatingGroupCount = new JSONObject();
+        repeatingGroupCount.put(KEY, widgetArgs.getJsonObject().get(KEY) + "_count");
+        repeatingGroupCount.put(OPENMRS_ENTITY_PARENT, "");
+        repeatingGroupCount.put(OPENMRS_ENTITY, "");
+        repeatingGroupCount.put(OPENMRS_ENTITY_ID, "");
+        repeatingGroupCount.put(TYPE, "");
+        repeatingGroupCount.put(TEXT, widgetArgs.getJsonObject().get(REFERENCE_EDIT_TEXT_HINT));
+        getStepFields(getJsonApi(widgetArgs).getStep(widgetArgs.getStepName())).put(repeatingGroupCount);
```

Comment:
1. Is an edge-case that can be addressed, although the fields with no values attached do not appear in the final event so not sure if that is necessary. It would only be a problem if the person navigated and edited the repeating group count. In which case, I believe the last recorded value would be picked up in the event?
2. This is not a widget, it's just a simple field that holds data.
3. Text can be used when extracting data and you need the question that was attached to this repeating group.
4. OpenSRP methods sometimes have weird constraints around fields missing, doesn't hurt to have it added and matching the widget definition as seen on a standard form.

Reply:
For 1, we can also add the element as a key, jsonobject pair.

Reply:
Added a check for concern 1.